--- arch/x86/realmode/rmpiggy.S.new	2022-04-21 09:00:39.636016815 -0400
+++ arch/x86/realmode/rmpiggy.S	2022-04-21 09:00:51.392134904 -0400
@@ -3,17 +3,17 @@
  * Wrapper script for the realmode binary as a transport object
  * before copying to low memory.
  */
-#include <linux/linkage.h>
-#include <asm/page_types.h>
-
-	.section ".init.data","aw"
-
-	.balign PAGE_SIZE
-
-SYM_DATA_START(real_mode_blob)
-	.incbin	"arch/x86/realmode/rm/realmode.bin"
-SYM_DATA_END_LABEL(real_mode_blob, SYM_L_GLOBAL, real_mode_blob_end)
-
-SYM_DATA_START(real_mode_relocs)
-	.incbin	"arch/x86/realmode/rm/realmode.relocs"
-SYM_DATA_END(real_mode_relocs)
+//#include <linux/linkage.h>
+//#include <asm/page_types.h>
+//
+//	.section ".init.data","aw"
+//
+//	.balign PAGE_SIZE
+//
+//SYM_DATA_START(real_mode_blob)
+//	.incbin	"arch/x86/realmode/rm/realmode.bin"
+//SYM_DATA_END_LABEL(real_mode_blob, SYM_L_GLOBAL, real_mode_blob_end)
+//
+//SYM_DATA_START(real_mode_relocs)
+//	.incbin	"arch/x86/realmode/rm/realmode.relocs"
+//SYM_DATA_END(real_mode_relocs)
