--- Makefile.flags	2022-11-01 18:18:43.006550705 -0400
+++ Makefile.flags.new	2022-11-01 18:20:21.794697686 -0400
@@ -108,12 +108,15 @@
 # gcc-4.2.1 fails if we try to feed C source on stdin:
 #  echo 'int main(void){return 0;}' | $(CC) $(CFLAGS) -lcrypt -o /dev/null -xc -
 # fall back to using a temp file:
+ifeq ($(BUSYBOX_NO_CRYPT_CHECK),1)
+else
 CRYPT_AVAILABLE := $(shell echo 'int main(void){return 0;}' >crypttest.c; $(CC) $(CFLAGS) -lcrypt -o /dev/null crypttest.c >/dev/null 2>&1 && echo "y"; rm crypttest.c)
 ifeq ($(CRYPT_AVAILABLE),y)
 LDLIBS += m crypt
 else
 LDLIBS += m
 endif
+endif
 
 ifeq ($(CONFIG_PAM),y)
 # libpam uses libpthread, so for static builds busybox must be linked to
